src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressPane.cs

```diff
@@ -70,65 +71,93 @@ class ProgressPane
                 int rows = tempProgressRegion.GetLength(0);
                 int cols = tempProgressRegion.GetLength(1);
 
+                if (ProgressNode.IsMinimalProgressRenderingEnabled())
+                {
+                    rows = _content.Length;
+                    cols = PSStyle.Instance.Progress.MaxWidth;
+                    if (cols > _bufSize.Width)
+                    {
+                        cols = _bufSize.Width;
+                    }
+                }
+
                 _savedCursor = _rawui.CursorPosition;
                 _location.X = 0;
 
-#if UNIX
-                _location.Y = _rawui.CursorPosition.Y;
-
-                // if cursor is not on left edge already move down one line
-                if (_rawui.CursorPosition.X != 0)
+                if (!Platform.IsWindows || ProgressNode.IsMinimalProgressRenderingEnabled())
                 {
-                    _location.Y++;
-                    _rawui.CursorPosition = _location;
-                }
+                    _location.Y = _rawui.CursorPosition.Y;
 
-                // if the cursor is at the bottom, create screen buffer space by scrolling
-                int scrollRows = rows - ((_rawui.BufferSize.Height - 1) - _location.Y);
-                if (scrollRows > 0)
-                {
-                    // Scroll the console screen up by 'scrollRows'
-                    var bottomLocation = _location;
-                    bottomLocation.Y = _rawui.BufferSize.Height;
-                    _rawui.CursorPosition = bottomLocation;
-                    for (int i = 0; i < scrollRows; i++)
+                    // if cursor is not on left edge already move down one line
+                    if (_rawui.CursorPosition.X != 0)
                     {
-                        Console.Out.Write('\n');
+                        _location.Y++;
+                        _rawui.CursorPosition = _location;
                     }
 
-                    _location.Y -= scrollRows;
-                    _savedCursor.Y -= scrollRows;
-                }
+                    // if the cursor is at the bottom, create screen buffer space by scrolling
+                    int scrollRows = rows - ((_rawui.BufferSize.Height - 1) - _location.Y);
+                    if (scrollRows > 0)
+                    {
+                        // Scroll the console screen up by 'scrollRows'
+                        var bottomLocation = _location;
+                        bottomLocation.Y = _rawui.BufferSize.Height;
+                        if (Platform.IsWindows)
+                        {
+                            bottomLocation.Y--;
```

Comment:
will add comment

Reply:
I checked this and it appears the reason I had this is on Unix, if you set the cursor.Y to >= WindowHeight (starting at index 0), it will automatically put it at the bottom of the screen.  However, on Windows, if cursor.Y is > WindowHeight, then you get an exception.  So the right behavior here isn't to special case Windows, but to always be WindowHeight-1.